indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorSpec.java

```diff
@@ -93,6 +96,7 @@ public SeekableStreamSupervisorSpec(
   )
   {
     this.ingestionSchema = checkIngestionSchema(ingestionSchema);
+    this.id = Preconditions.checkNotNull(Configs.valueOrDefault(id, ingestionSchema.getDataSchema().getDataSource()), "spec id cannot be null!");
```

Comment:
We should check for null but not default to datasource here.
The sub classes should pass in the correct value.